protoc-gen-openapiv2/options/openapiv2.proto

```diff
@@ -505,6 +506,7 @@ message JSONSchema {
   reserved 43 to 45;
   // Items in `enum` must be unique https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.5.1
   repeated string enum = 46;
+  map<string, google.protobuf.Value> extensions = 47;
```

Comment:
Which section within the swagger spec does this map to: https://swagger.io/docs/specification/2-0/swagger-extensions/? Just want to make sure we're not adding these somewhere they don't belong.

Reply:
If I'm understanding correctly, this is used for [`FieldOptions`](https://github.com/grpc-ecosystem/grpc-gateway/blob/185126621f298395754503abc6460e5d9e1f62b9/protoc-gen-openapiv2/options/annotations.proto#L29) annotations, I think this maps to the swagger schema object which supports extensions according to: https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#patterned-objects-9. Example:

```
message AMessage {
  string id = 1 [
    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
      extensions: {
        key: "x-something-something";
        value {
          string_value: "yadda";
        }
      }
    }
  ]
};
```

However, in revisiting this I've realised that the `JSONSchema` defined here is also part of the `Schema` type defined above, so it looks like it would be possible to only add the `extensions` field in this `JSONSchema` message and use in schema annotations like this:

```
message AMessage {
  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_schema) = {
    json_schema: {
      extensions: {
        key: "x-something-something";
        value {
          string_value: "yadda";
        }
      }
    }
  };
}
```

Reply:
That sounds reasonable to me